kolibri/core/content/urls.py

```diff
@@ -16,4 +17,5 @@
         {},
         "downloadcontent",
     ),
+    url(r"^viewcontent$", ContentPermalinkRedirect.as_view(), name="contentpermalink"),
```

Comment:
What's the utility of putting this in `core` rather than keeping it namespaced under `learn`?

I don't have a good sense of the use-cases we expect to support by allowing any plugin to implement a `ContentNodeDisplayHook`-defined content view redirect. I'm also interested the precedent we're establishing here for other similar redirects in the future.

A quick list of semi-rhetorical questions...

* Is "view content" meant to imply a Learner display versus e.g. viewing analytics or metadata?
* If it's always Learn, does it imply a specific learning display modality? (self-contained/isolated vs contextualized in a channel, lesson, etc)?
* Will we ever want to parameterize 'view content' further to provide more specific instructions?
* Do we expect to re-use this pattern for other content or DB model types, e.g. channel, facility, user, etc?
* Are there situations where multiple plugins might legitimately register handlers for the same or similar URL patterns?

Reply:
> Is "view content" meant to imply a Learner display versus e.g. viewing analytics or metadata?

Yes

> If it's always Learn, does it imply a specific learning display modality? (self-contained/isolated vs contextualized in a channel, lesson, etc)?

No - it is up to the individual plugin to determine how the redirect should be resolved. In the case of learn, if we wanted to get smart, we could try to see if that content item existed in any lessons currently assigned to the logged in user and redirect based on that. I have not done that, but the hook is unopinionated about whether this should happen or not.

> Will we ever want to parameterize 'view content' further to provide more specific instructions?

Yes - future versions could take additional GET params, but allow the fallback behaviour encoded here if that Kolibri version does not support these, or the higher specificity is not locally available.

> Do we expect to re-use this pattern for other content or DB model types, e.g. channel, facility, user, etc?

No - I don't think so. Content is different because it plays a very different role across the Kolibri ecosystem and appears consistently in many different Kolibri installations. It is also fundamentally public, whereas most of the other data in Kolibri is private to more or lesser a degree.

Channels might be one other place where we might offer a universal redirect, but I don't have a use case in mind for why yet.

> Are there situations where multiple plugins might legitimately register handlers for the same or similar URL patterns?

Assuming I am understanding this correctly - this is possible, but as all registered plugin urls are namespaced under the plugin name, the plugins would have to be named identically for this to occur (but we should also complain noisily - we don't currently - if this happens).

Reply:
> Assuming I am understanding this correctly 

This was essentially the same question as https://github.com/learningequality/kolibri/pull/5641#discussion_r292145083

basically, if multiple plugins have registered the same pattern, how do we break the tie?

Reply:
Ah, specifically for the redirects? There is no mechanism atm, no. So putting it as only one would be the quick fix for that.

Reply:
> Channels might be one other place where we might offer a universal redirect, but I don't have a use case in mind for why yet.

And that could probably already be handled using this right? Just using the root node ID (which is generally the same as the channel ID)?